internal/csi-addons/rbd/volumegroup.go

```diff
@@ -117,6 +117,21 @@ func (vs *VolumeGroupServer) CreateVolumeGroup(
 
 	log.DebugLog(ctx, "VolumeGroup %q has been created: %+v", req.GetName(), vg)
 
+	// extract the flatten mode
+	flattenMode, err := getFlattenMode(ctx, req.GetParameters())
+	if err != nil {
+		return nil, err
```

Comment:
This needs to return a grpc error. `getFlattenMode` is not appropriate to do so.

Reply:
getFlattenMode is returning grpc error same is already used in replication.